src/daemon/rpmostreed-deployment-utils.c

```diff
@@ -385,16 +397,23 @@ add_all_commit_details_to_vardict (OstreeDeployment *deployment,
 
   g_autofree gchar *refspec_owned = NULL;
   gboolean refspec_is_ostree = FALSE;
+  RpmOstreeRefspecType refspec_type;
   if (!refspec)
     {
       g_autoptr(RpmOstreeOrigin) origin =
         rpmostree_origin_parse_deployment (deployment, error);
       if (!origin)
         return FALSE;
-      RpmOstreeRefspecType refspec_type;
       refspec = refspec_owned = rpmostree_origin_get_full_refspec (origin, &refspec_type);
-      refspec_is_ostree = (refspec_type == RPMOSTREE_REFSPEC_TYPE_OSTREE);
     }
+  else
+    {
+      const char *refspec_remainder = NULL;
+      if (!rpmostree_refspec_classify (refspec, &refspec_type, &refspec_remainder, error))
+        return FALSE;
+      refspec = refspec_remainder;
+    }
+  refspec_is_ostree = (refspec_type == RPMOSTREE_REFSPEC_TYPE_OSTREE);
```

Comment:
`refspec_type` is uninitialized if we take the first branch.

Reply:
It's filled by `rpmostree_origin_get_full_refspec` no?

Reply:
Sorry, you're right!